third_party/gpus/cuda_configure.bzl

```diff
@@ -158,20 +147,63 @@ def _cudnn_install_basedir(repository_ctx):
   return cudnn_install_path
 
 
-def _cuda_version(repository_ctx):
+def _lib_version(lib_name, repository_ctx, cpu_value, base_dir):
+  lib = _find_cuda_lib(lib_name, repository_ctx, cpu_value, base_dir, "")
+  lib_path = lib.path
+  basename = repository_ctx.path(lib_path).basename
+  if cpu_value == "Linux":
+    if len(basename) == len("lib%s.so" % lib_name):
+      auto_configure_fail("Could not find versioned %s lib. Found: %s" %
+                          (lib_name, lib_path))
+    full_version = basename.replace("lib%s.so." % lib_name, "")
+  elif cpu_value == "Darwin":
+    if len(basename) == len("lib%s.dylib" % lib_name):
+      auto_configure_fail("Could not find versioned %s lib. Found: %s" %
+                          (lib_name, lib_path))
+    full_version = (
+        basename.replace("lib%s" % lib_name, "").replace(".dylib", ""))
+  elif cpu_value == "Windows":
+    if len(basename) == len("%s.dll" % lib_name):
+      auto_configure_fail("Could not find versioned %s lib. Found: %s" %
+                          (lib_name, lib_path))
+    full_version = basename.replace(lib_name, "").replace(".dll", "")
```

Comment:
Seems like this can be factorized:

```
EXTENSIONS = {"Linux": "so", "Darwin": "dylib", "Windows": "dll"}
PREFIXES = {"Linux": "lib", "Darwin": "lib", "Windows": ""}
if not cpu_value in EXTENSIONS.keys():
 auto_configure_fail("Not supported CPU value %s" % cpu_value)
expected_basename = "%s%s" % (PREFIXES[cpu_value], lib_name)
extension = EXTENSIONS[cpu_value]
expected_name = "%s.%s" % (expected_basename, extension)
if len(basename) == len(expected_name):
    auto_configure_fail("Could not find versioned %s lib. Found: %s" % (lib_name, lib_path))
full_version = basename.replace(expected_basename, "").replace("." + extension, "")